accepted/00000-enable-ansible-control-node-in-registered-system.md

```diff
@@ -0,0 +1,136 @@
+
+At this point, Uyuni doesn't know how to reach those systems, we only know their "fqdns" according to the Ansible inventory, but the Ansible inventory in the control node should contain information about how to reach them via ssh. The idea here is the following:
+
+In order to bootstrap and trigger the registration of these systems, we have two alternatives:
+
+#### Use "salt-ssh" from Uyuni directly to the "fqdns" of the Ansible managed clients:
+
+* We previously need to fetch the corresponding SSH key that is used to connect to this particular host by Ansible.
+* Then we would reuse this key as SSH credentials to trigger the bootstrap on that host.
+* Downside: Uyuni needs to have direct access to the Ansible managed clients to bootstrap
+
+#### Use "salt-ssh" from Uyuni using the Ansible control node as SSH proxy:
+
+* We do not necessary need to fetch the corresponding SSH key, but we need to have the Uyuni SaltSSH deployed on the Ansible control node to access it
+* The SSH key to connect the Ansible managed client referenced in the Ansible inventory is used to build the ProxyCommand to use in "salt-ssh".
+
+NOTE: It needs to be clarified in the documentation that `ansible_ssh_private_key_file` needs to be defined in the Ansible inventory of your Ansible control node.
+
+* Once we have this, we can trigger the bootstrap of those systems by calling "salt-ssh" like this:
+
+```console
+# salt-ssh --ssh-option='ProxyCommand="/usr/bin/ssh -i /srv/susemanager/salt/salt_ssh/mgr_ssh_id -o StrictHostKeyChecking=no -o User=root -W %h:%p uyuni-ansible-control-node.tf.local"' --roster=ansible --roster-file=/var/cache/ansible/ansible-inventory.yaml -N webserver1 state.apply certs,bootstrap pillar='{"mgr_server": "uyuni-srv.tf.local", "minion_id": "uyuni-ansible-sles15sp1-2.tf.local"}'
+```
+In esence, we apply the same states that we do at the time of Boostrapping a new minion via the UI, passing the necessary information as pillar data.
+
+* Another example of usage (without using the ansible roster/targetting):
+
+```console
+# salt-ssh --ssh-option='ProxyCommand="/usr/bin/ssh -i PATH_TO_SALT_SSH_KEY_IN_THE_UYUNI_SERVER -o StrictHostKeyChecking=no -W %h:%p uyuni-stable-ansible-controller.tf.local /usr/bin/ssh -i PATH_TO_THE_SSH_KEY_IN_ANSIBLE_CONTROL_NODE uyuni-stable-ansible-opensuse152-2.tf.local" -o StrictHostKeyChecking=no' uyuni-stable-ansible-opensuse152-2.tf.local test.ping
+```
+
+
+### Triggering playbook executions in an Ansible control node
+
+After exploring the Ansible control node, inventory and playbooks, we can allow Uyuni to trigger the execution of one of those playbooks via a simple Salt state to execute in that control node:
+
+```yaml
+execute_ansible_playbook:
+  ansible.playbooks:
+    - name: /some/path/in/my/ansible/control/node/playbook_1.yaml
+```
+
+We would need to create a new type of Action in Uyuni for "running a playbook". We want this action to appear in the system event history for the Ansible control node.
```

Comment:
It is my understanding that we do not intend to store the list of playbooks in Postgres at least for now. The list is retrieved live at page load, displayed as a list with a button to run each, and once the button is click the action is scheduled.

Correct?

Reply:
That's correct, at least for now.

Beside of listing the playbooks, we probably also want the user can display the content of the playbook (as readonly) before triggering it clicking the button, but this can be done in live without storing any content in the DB:

```console
# salt ANSIBLE_CONTROL_NODE cp.get_file_str /path/to/the/playbook
```